code/modules/power/singularity/singularity.dm

```diff
@@ -446,3 +446,136 @@
 	explosion(src.loc,(dist),(dist*2),(dist*4))
 	qdel(src)
 	return(gain)
+
+//////////////////////////////////////
+// DEAD CHAT CONTROLLED SINGULARITY //
+//////////////////////////////////////
+
+/datum/deadchat_listener/singularity_listener
+	name = "deadchat-controlled singularity listener"
+	var/obj/singularity/deadchat_controlled/parent
+
+/datum/deadchat_listener/singularity_listener/message_event(ckey, message) 
+	parent.process_deadchat(ckey,message)
+
+/obj/singularity/deadchat_controlled
+	desc = "A gravitational singularity, but it seems odd..."
+	move_self = FALSE
+	/// The mode how the deadchat movement functions, possible modes are Democracy and Anarchy.
+	var/deadchat_mode = "Democracy"
+	/// How long it takes until a move can be done by a single person in Anarchy mode.
+	var/anarchy_cooldown = 20 SECONDS
+	/// How long it takes until a new voting cycle beings in Democracy mode.
+	var/democracy_cooldown = 12 SECONDS
+	/// List of ckeys that have voted already and/or are on cooldown.
+	var/list/ckey_to_cooldown = list()
+	/// What inputs will the singularity be looking for in the messages.
+	var/list/inputs = list("up", "down", "left", "right")
+	/// deadchat_listener datum for the singularity.
+	var/datum/deadchat_listener/singularity_listener/listener
+
+/obj/singularity/deadchat_controlled/Initialize(mapload, starting_energy = 50)
+	..()
+	listener = new /datum/deadchat_listener/singularity_listener
+	listener.parent = src
+	GLOB.deadchat_listeners += listener
+	
+/obj/singularity/deadchat_controlled/Destroy()
+	..()
+	var/message = "<span class='deadsay italics bold'>The deadchat controlled singularity has been destroyed.</span>"
+	deadchat_broadcast(message)
+	GLOB.deadchat_listeners -= listener
+	qdel(listener)
+	
+/obj/singularity/deadchat_controlled/proc/process_deadchat(ckey, message)
+	if(deadchat_mode == "Anarchy")
+		var/cooldown = ckey_to_cooldown[ckey]
+		if(!cooldown)
+			ckey_to_cooldown[ckey] = NONE
+			cooldown = 0
+		if(cooldown > 0)
+			return
+		var/direction
+		message = lowertext(message)
+		switch(message)
```

Comment:
Personal preference but if you make a list of strings - defines you don't have to do this switch